### PR TITLE
feat(metadata-views): Save inline edited metadata using fields feature

### PR DESCRIPTION
**Pull request by manishPh**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @manishPh has signed the CLA. Thanks for the pull request!

File: src/elements/content-explorer/ContentExplorer.js [hunk 1]
```diff
@@ -1431,6 +1433,58 @@ class ContentExplorer extends Component<Props, State> {
...
+                });
+            }
+            return clonedItem;
+        });
+
+        this.setState({
+            currentCollection: {
+                items: updatedItems,
+                nextMarker,
+                percentLoaded: 100,
```
Comment (swfree) [inline src/elements/content-explorer/ContentExplorer.js]: do you also need to reset isLoading state to false, or is updating percentLoaded sufficient?

File: src/elements/content-explorer/ContentExplorer.js [hunk 2]
```diff
@@ -1431,6 +1433,58 @@ class ContentExplorer extends Component<Props, State> {
...
+            oldValue,
+            newValue,
+            () => {
+                this.updateMetadataSuccessCallback(item, field, newValue);
+            },
+            this.errorCallback,
+        );
+    };
+
+    updateMetadataSuccessCallback = (item: BoxItem, field: string, newValue: ?MetadataFieldValue): void => {
```
Comment (swfree) [inline src/elements/content-explorer/ContentExplorer.js]: could you add a unit test for this?

File: src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js [hunk 3]
```diff
@@ -122,11 +184,25 @@ describe('features/metadata-based-view/MetadataBasedItemList', () => {
...
-                instance.handleMouseEnter(rowIndex);
+                instance.handleMouseEnter(columnIndex, rowIndex);
             }
 
             const data = instance.getGridCellData(columnIndex, rowIndex);
-            expect(data).toEqual(cellData);
+            if (columnIndex < 2) {
+                expect(data).toEqual(cellData);
+                // i.e. FileIcon and FileName columns
+                expect(data).toEqual(cellData);
```
Comment (swfree) [inline src/features/metadata-based-view/__tests__/MetadataBasedItemList.test.js]: duplicate line to line 192

File: src/elements/content-explorer/ContentExplorer.js (hunk 1 above)
Comment (manishPh) [reply src/elements/content-explorer/ContentExplorer.js]: Looks like I don't need `isLoading`. This flag is used to control dialog(modal) behavior. `percentLoaded` is sufficient.

**Pull request merged**